app.js

```diff
@@ -0,0 +1,81 @@
+  var calc_choice = prompt("Which calculator will you use?\n(a)Basic\n(b)Advanced\n(c)BMI\n(d)Travel Fare");
+
+  if (calc_choice=='a'||calc_choice=='b') {
+    var num1 = parseInt(prompt("Enter the first number."));
+    var num2 = parseInt(prompt("Enter the second number."));
+  }
+
+  switch (calc_choice) {
+    case 'a':
+      var operator_choice = prompt("Your numbers are "+num1+" and "+num2+"\nWhich operation will you use?\n(a)Addition\n(b)Subtraction\n(c)Multiplication\n(d)Division");
+      break;
+    case 'b':
+      var operator_choice = prompt("Your numbers are "+num1+" and "+num2+"\nWhich operation will you use?\n(a)Power\n(b)Square Root");
+    case 'c':
+      var operator_choice = prompt("Which units would you like?\n(a)Metric\n(b)Imperial");
+      break;
+    case 'd':
+      var distance = parseInt(prompt("Enter distance travelled in miles."));
+      var efficiency = parseInt(prompt("Enter fuel efficiency in mpg."));
+      var cost = parseInt(prompt("Enter cost per gallon in pounds."));
+      var speed = parseInt(prompt("Enter speed in mph."));
+      if (speed>60) {
+        efficiency -= ((speed-60)*2)
+        if (efficiency<=20) {
+          efficiency = 20;
+        }
+      }
+      break;
+    default:
+      alert("Nothing happened.");
```

Comment:
Good use of a default message testing for invalid input, but I would make it something more specific to the situation e.g. "Please enter a valid input"